static/app/utils/replays/useReplayData.tsx

```diff
@@ -109,55 +114,40 @@ function useReplayData({eventSlug, orgId}: Options): Result {
         const response = await api.requestPromise(
           `/api/0/projects/${orgId}/${projectId}/events/${eventId}/attachments/${attachment.id}/?download`
         );
-        return JSON.parse(response).events as eventWithTime;
+        return JSON.parse(response) as ReplayAttachment;
       })
     );
-    return attachments.flat();
-  }, [api, eventId, orgId, projectId]);
-
-  const fetchReplayEvents = useCallback(async () => {
-    const replayEventList = await api.requestPromise(
-      `/organizations/${orgId}/eventsv2/`,
-      {
-        query: {
-          statsPeriod: '14d',
-          project: [],
-          environment: [],
-          field: ['timestamp', 'replayId'],
-          sort: 'timestamp',
-          per_page: 50,
-          query: ['transaction:sentry-replay-event', `replayId:${eventId}`].join(' '),
-        },
-      }
-    );
 
-    return Promise.all(
-      replayEventList.data.map(
-        event =>
-          api.requestPromise(
-            `/organizations/${orgId}/events/${generateEventSlug(event)}/`
-          ) as Promise<Event>
-      )
-    );
-  }, [api, eventId, orgId]);
+    // ReplayAttachment[] => ReplayAttachment (merge each key of ReplayAttachment)
+    const result = attachments.reduce((acc, attachment) => {
+      Object.entries(attachment).forEach(([key, value]) => {
+        if (!acc[key]) {
+          acc[key] = [];
+        }
+        acc[key] = acc[key].concat(value);
+      });
+      return acc;
+    }, {}) as ReplayAttachment;
+    return result;
```

Comment:
I was going to say let's put the reduce call into the reader class, but then I was thinking nope.
If the api in the future makes it easy to pluck out the fields: `*.replaySpans` and so on, then its fine to leave here. Single responsibility of this class seems intact.

It's a generic method though, so it could be a candidate to unittest if exported